### ISSUE TITLE
Converting circular structure to JSON warning

### ISSUE THREAD

**Issue by iamzifei** [aksonov/react-native-router-flux#383]
I'm just trying v3 with the basic example provided on README, my index.ios.js looks like bellowing:

```
class ReactComponent extends React.Component {
  render() {
    return <Router>
          <Scene key="app">
            <Scene key="test" component={Test}/>
            <Scene key="home" component={Home}/>
          </Scene>
        </Router>
  }
}
```
And I have basic Test and Home component like bellowing:
```
class Home extends React.Component {
  render(){
    return (
      <View>
        <Text>Home screen</Text>
      </View>
    );
  }
}

module.exports = Home;
```

I'm having the following error `Converting circular structure to JSON` in line 53 in Actions.js.

Also would like to see an example using the new `Switch` for authentication.

**Comment by JamesWatling** [comment]
Hi @iamzifei, I also had this problem, and was fixed by adding a `key="root"` to the parent scene.

ie. 

```
      <Provider store={this.store}>
        <Router navigationBarStyle={styles.navbar} leftButtonTextStyle={styles.backButton}>
            <Scene key="root" navigationBarStyle={styles.navbar}>
              <Scene key={Routes.LAUNCH_SCREEN} initial={!this._loggedIn()} component={LaunchContainer} type='replace' />
          </Scene>
        </Router>
      </Provider>
```

**Comment by iamzifei** [comment]
Hi @JamesWatling , thanks for the reply. as you can see in my code, I have key='app' in parent scene. but still get this error. However when I clear everything (including node_modules) and re-run it again it works. Not sure why though.

**Comment by sashafklein** [comment]
Already had the ` key="root" ` set, and deleting node_modules didn't do anything for me, unfortunately. Looks like this:

```
    return (
      <Router>
        <Scene hideNavBar={true} key="root">
          <Scene key="messages" component={ Messages } initial={ true } title="Messages" />

          <Scene key="message" component={ Message } title="Message" />
          <Scene key="idCard" component={ IdCard } title="IdCard" />
        </Scene>
      </Router>
    );
```

The root page is just a hello world example at the moment.

**Comment by aksonov** [comment]
Should be fixed with 3.1.6
